zarr/compat.py

```diff
@@ -4,19 +4,26 @@
 import sys
 
 
+import numpy as np
+
+
 PY2 = sys.version_info[0] == 2
 
 
+numpy_integer_types = np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16, np.uint32, \
+                      np.uint64
+
+
 if PY2:  # pragma: no cover
 
     text_type = unicode
     binary_type = str
-    integer_types = (int, long)
+    integer_types = (int, long) + numpy_integer_types
     reduce = reduce
 
 else:
 
     text_type = str
     binary_type = bytes
-    integer_types = int,
+    integer_types = (int,) + numpy_integer_types
```

Comment:
Were we using this conversion or type checking? If the latter, [`numbers.Integral`]( https://docs.python.org/2/library/numbers.html#numbers.Integral ) works well for type checking all integers including NumPy typed ones.